online-fishing-store/src/app/pages/main-page/products/product-items/product-item.component.ts

```diff
@@ -9,12 +10,17 @@ import { Product } from 'src/app/interfaces/product';
 export class ProductItemComponent implements OnInit {
   @Input() public product: Product;
 
+  constructor(
+    public shoppingCartService:ShoppingCartService
+    ) {}
+
   public productAltImgAttribute: string = 'photo';
   public shoppingCartButtonText: string = 'В КОРЗИНУ';
 
-  public addShoppingCart(): void {}
-
-  constructor() {}
+  public addShoppingCart(event: Event, product: Product): void {
+    this.shoppingCartService.addShoppingCart(product)
+    event.stopPropagation();
```

Comment:
For what it??
I don't understand why you use it here.
For what you need here event parameter??